src/components/Request/Table.js

```diff
@@ -0,0 +1,242 @@
+        TableContainer, 
+        TableHead, 
+        Paper,
+        TableFooter,
+        IconButton,
+        TablePagination,
+        Fab,
+        Typography,
+        Tooltip,
+        Checkbox
+    } 
+from '@material-ui/core';
+import AddIcon from '@material-ui/icons/Add';
+import FilterListIcon from '@material-ui/icons/FilterList';
+import { KeyboardArrowLeft, KeyboardArrowRight,} from '@material-ui/icons';
+import DeleteIcon from '@material-ui/icons/Delete';
+import FirstPageIcon from '@material-ui/icons/FirstPage';
+import LastPageIcon from '@material-ui/icons/LastPage';
+import { useTheme, withStyles } from '@material-ui/core/styles'
+import { rows } from '../../helpers/tableData';
+import { useStyles, useStyles1 } from '../../shared/styles/TableStyles';
+import { id } from '../../helpers/tableData';
+
+
+//pagination actions
+export const TablePaginationActions = (props) => {
+    const classes = useStyles1();
+    const theme = useTheme();
+    const { count, page, rowsPerPage, onChangePage } = props;
+  
+    const handleFirstPageButtonClick = (event) => {
+      onChangePage(event, 0);
+    };
+  
+    const handleBackButtonClick = (event) => {
+      onChangePage(event, page - 1);
+    };
+  
+    const handleNextButtonClick = (event) => {
+      onChangePage(event, page + 1);
+    };
+  
+    const handleLastPageButtonClick = (event) => {
+      onChangePage(event, Math.max(0, Math.ceil(count / rowsPerPage) - 1));
+    };
+  
+    return (
+      <div className={classes.root}>
+        <IconButton
+          onClick={handleFirstPageButtonClick}
+          disabled={page === 0}
+          aria-label="first page"
+          
+        >
+          {theme.direction === 'rtl' ? <LastPageIcon /> : <FirstPageIcon />}
+        </IconButton>
+        <IconButton onClick={handleBackButtonClick} disabled={page === 0} aria-label="previous page">
+          {theme.direction === 'rtl' ? <KeyboardArrowRight /> : <KeyboardArrowLeft />}
+        </IconButton>
+        <IconButton
+          onClick={handleNextButtonClick}
+          disabled={page >= Math.ceil(count / rowsPerPage) - 1}
+          aria-label="next page"
+        >
+          {theme.direction === 'rtl' ? <KeyboardArrowLeft /> : <KeyboardArrowRight />}
+        </IconButton>
+        <IconButton
+          onClick={handleLastPageButtonClick}
+          disabled={page >= Math.ceil(count / rowsPerPage) - 1}
+          aria-label="last page"
+        >
+          {theme.direction === 'rtl' ? <FirstPageIcon /> : <LastPageIcon />}
+        </IconButton>
+      </div>
+    );
+  }
+  
+  TablePaginationActions.propTypes = {
+    count: PropTypes.number.isRequired,
+    onChangePage: PropTypes.func.isRequired,
+    page: PropTypes.number.isRequired,
+    rowsPerPage: PropTypes.number.isRequired,
+  };
+
+  const StyledTableRow = withStyles((theme) => ({
+    root: {
+      "&:hover": {
+        backgroundColor: theme.palette.action.hover,
+      },
+    },
+  }))(TableRow);
+  
```

Comment:
//TODO
For maintainability, modularity and reusability make a reusable table component that all the pages displaying a table can just reuse to avoid code redundancy

Reply:
@Malaba6  kindly re-review the PR